packages/app/src/heartbeatService.ts

```diff
@@ -82,28 +89,28 @@ export class HeartbeatServiceImpl implements HeartbeatService {
     const userAgent = platformLogger.getPlatformInfoString();
     const date = getUTCDateString();
     if (this._heartbeatsCache === null) {
-      await this._heartbeatsCachePromise;
+      this._heartbeatsCache = await this._heartbeatsCachePromise;
     }
-    let heartbeatsEntry = this._heartbeatsCache!.find(
-      heartbeats => heartbeats.userAgent === userAgent
-    );
-    if (heartbeatsEntry) {
-      if (heartbeatsEntry.dates.includes(date)) {
-        // Only one per day.
-        return;
-      } else {
-        // Modify in-place in this.heartbeatsCache
-        heartbeatsEntry.dates.push(date);
-      }
+    if (
+      this._heartbeatsCache.some(
+        singleDateHeartbeat => singleDateHeartbeat.date === date
+      )
+    ) {
+      // Do not store a heartbeat if one is already stored for this day.
+      return;
     } else {
-      // There is no entry for this Firebase user agent. Create one.
-      heartbeatsEntry = {
-        userAgent,
-        dates: [date]
-      };
-      this._heartbeatsCache!.push(heartbeatsEntry);
+      // There is no entry for this date. Create one.
+      this._heartbeatsCache!.push({ date, userAgent });
```

Comment:
Can you get rid of the `!` assertion now that it's explicitly set on line 92?

Reply:
Thanks, done.